spec/models/grace_period_submission_rule_spec.rb

```diff
@@ -5,6 +5,28 @@
     expect(rule.save).to be_truthy
   end
 
+  context 'When the student did not submit any assignment files' do
+    describe '#apply_submission_rule' do
+      before :each do
+        @group = create(:group)
```

Comment:
Instead of setting all of these new objects as instance attributes in the `before :each` method, can we write these as `let` statements?

Reply:
Yes we can but i thought If we are going add more test under 'when the student did not submit any assignment files' it would be useful to set everything under before :each method.